Kotlin/src/test/kotlin/com/gildedrose/GildedRoseTest.kt

```diff
@@ -153,7 +153,65 @@ internal class GildedRoseTest {
+            ItemType("Aged Brie", 10, 10, false),
+            ItemType("Sulfuras, Hand of Ragnaros", 0, 80, false),
+            ItemType("Backstage passes to a TAFKAL80ETC concert", 12, 10, false),
+            ItemType("Backstage passes to a TAFKAL80ETC concert", 8, 10, false),
+            ItemType("Backstage passes to a TAFKAL80ETC concert", 2, 10, false),
+            ItemType("Backstage passes to a TAFKAL80ETC concert", -2, 0, false))
+
+        val conjuredItems = listOf(
+            ItemType("Basic", 10, 10, true),
+            ItemType("BasicPastSellBy", -1, 10, true),
+            ItemType("Aged Brie", 10, 10, true),
+            ItemType("Sulfuras, Hand of Ragnaros", 0, 80, true),
+            ItemType("Backstage passes to a TAFKAL80ETC concert", 12, 10, true),
+            ItemType("Backstage passes to a TAFKAL80ETC concert", 8, 10, true),
+            ItemType("Backstage passes to a TAFKAL80ETC concert", 2, 10, true),
+            ItemType("Backstage passes to a TAFKAL80ETC concert", -2, 0, true))
+
+        val app = GildedRose(items)
+        val conjuredApp = GildedRose(conjuredItems)
 
+        app.updateQuality()
+        conjuredApp.updateQuality()
+
+        val degradeRates = LinkedHashMap<String, Int>()
+        degradeRates["Basic"] = 10 - app.items[0].quality
+        degradeRates["BasicPastSellBy"] = 10 - app.items[1].quality
+        degradeRates["Aged Brie"] = 10 - app.items[2].quality
+        degradeRates["Sulfuras"] = 80 - app.items[3].quality
+        degradeRates["Backstage12Days"] = 10 - app.items[4].quality
+        degradeRates["Backstage8Days"] = 10 - app.items[5].quality
+        degradeRates["Backstage2Days"] = 10 - app.items[6].quality
+        degradeRates["BackstageAfterConcert"] = 0 - app.items[7].quality
+
+        val conjuredDegradeRates = LinkedHashMap<String, Int>()
+        conjuredDegradeRates["Basic"] = 10 - conjuredApp.items[0].quality
+        conjuredDegradeRates["BasicPastSellBy"] = 10 - conjuredApp.items[1].quality
+        conjuredDegradeRates["Aged Brie"] = 10 - conjuredApp.items[2].quality //TODO: implement conjured aged brie and backstage passes (int will not work)
+        conjuredDegradeRates["Sulfuras"] = 80 - conjuredApp.items[3].quality
+        conjuredDegradeRates["Backstage12Days"] = 10 - conjuredApp.items[4].quality
+        conjuredDegradeRates["Backstage8Days"] = 10 - conjuredApp.items[5].quality
+        conjuredDegradeRates["Backstage2Days"] = 10 - conjuredApp.items[6].quality
+        conjuredDegradeRates["BackstageAfterConcert"] = 0 - conjuredApp.items[7].quality
+
+        assertEquals(conjuredDegradeRates["Basic"], 2 * degradeRates["Basic"]!!)
+        assertEquals(conjuredDegradeRates["BasicPastSellBy"], 2 * degradeRates["BasicPastSellBy"]!!)
+        assertEquals(conjuredDegradeRates["Sulfuras"], 2 * degradeRates["Sulfuras"]!!)
+        //assertEquals(conjuredDegradeRates["Backstage12Days"], 2 * degradeRates["Backstage12Days"]!!)
```

Comment:
Why are these commented-out lines left in the commit?

If you intend to re-add them then it's best to do so before opening the PR.
If you don't intend to re-add them, just delete them please :)